src/main/kotlin/pl/org/pablo/slack/money/money/MoneyServiceImpl.kt

```diff
@@ -32,67 +33,88 @@ class MoneyServiceImpl(
         updateBalance(from, to, value)
     }
 
+    /**
+     * Update the state of the debt between the to given users by the given value.
+     *
+     * @param from the user which owes money
+     * @param to the user which is owed to
+     * @param value the amount of money which is owed
+     */
     private fun updateBalance(from: UserEntity, to: UserEntity, value: Int) {
-        var remainingValue = optimizeBetween(from, to, value)
+        var remainingValue = reduceDebt(from, to, value)
         if (remainingValue == 0) {
             return
         }
-        // Check if I own anyone money and then move this dept
+        // Check if I owe anyone money and then move this debt
         if (from.toPay.isNotEmpty()) {
-            val res: Map<Boolean, List<BalanceRelationship>> = from.toPay.stream()
-                    .sorted { x, y -> Integer.compare(x.value, y.value) }
-                    .collect(Collectors.groupingBy { to.toPay.any { o -> it.receiver.id == o.receiver.id } })
+            val res = from.toPay.sortedBy { it.value }
```

Comment:
If you have two chained operation on collection it's might be wise to use `asSequence()` method before `sortedBy`.

Reply:
Done.